com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/AbstractSubProcessMonitorStep.java

```diff
@@ -64,7 +76,21 @@ private ErrorType getSubProcessErrorType(HistoricProcessInstance subProcess) {
     protected abstract ExecutionStatus onAbort(DelegateExecution context, ErrorType errorType) throws MonitoringException;
 
     protected ExecutionStatus onSuccess(DelegateExecution context) {
+        duplicateProcessVariablesFromSubProcess(context);
         return ExecutionStatus.SUCCESS;
     }
 
+    private void duplicateProcessVariablesFromSubProcess(DelegateExecution context) {

```

Comment:
The name of the method is a little bit tricky. Could we rename it?

Reply:
Done